backend/baseclass.pm

```diff
@@ -856,6 +856,13 @@ sub check_asserted_screen {
             backend::baseclass::write_crash_file();
             bmwqemu::mydie "assert_screen fails, but we detected a timeout in the process, so we abort";
         }
+        elsif (!$self->assert_screen_check) {
+            my @unregistered_needles = grep { $_->{unregistered} } @{$self->assert_screen_needles};
+            my ($foundneedle, $candidates) = $img->search(\@unregistered_needles, 0, 1, undef);
+            # the best here is still a failure, as unregistered
+            push(@$failed_candidates, $foundneedle) if $foundneedle;
+            push(@$failed_candidates, @$candidates);
```

Comment:
well, that's basically what I wanted to say today in the call. Just add some `diag` and we already have something useful :-)